app/models/topic_hot_score.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+class TopicHotScore < ActiveRecord::Base
+  belongs_to :topic
+
+  DEFAULT_BATCH_SIZE = 1000
+
+  def self.update_scores(max = DEFAULT_BATCH_SIZE)
+    # score is
+    # (total likes - 1) / (age in hours + 2) ^ gravity
```

Comment:
Should the number of recent replies also influence the score?

Reply:
latest iteration handles this :hugs: